core/iwasm/libraries/libc-uvwasi/libc_uvwasi_wrapper.c

```diff
@@ -72,9 +69,24 @@ typedef struct iovec_app {
     uint32 buf_len;
 } iovec_app_t;
 
+typedef struct WASIContext {
+    uvwasi_t uvwasi;
+    uint32_t exit_code;
+} WASIContext;
```

Comment:
That's probably out of scope of this PR as it needs a bit larger refactoring, but I think we should strive to have one declaration per context, instead of duplicating it. I've opened an issue to track it: https://github.com/bytecodealliance/wasm-micro-runtime/issues/1749

Reply:
i tend to agree. but it's how this code base is currently structured.